linera-sdk/src/test/chain.rs

```diff
@@ -89,6 +93,103 @@ impl ActiveChain {
         self.key_pair = key_pair
     }
 
+    /// Reads the current shared balance available to all of the owners of this microchain.
+    pub async fn chain_balance(&self) -> Amount {
+        let query = Query::System(SystemQuery);
+
+        let QueryOutcome { response, .. } = self
+            .validator
+            .worker()
+            .query_application(self.id(), query)
+            .await
+            .expect("Failed to query chain's balance");
+
+        let QueryResponse::System(SystemResponse { balance, .. }) = response else {
+            panic!("Unexpected response from system application");
+        };
+
+        balance
+    }
+
+    /// Reads the current account balance on this microchain of an [`AccountOwner`].
+    pub async fn owner_balance(&self, owner: &AccountOwner) -> Option<Amount> {
+        let chain_state = self
+            .validator
+            .worker()
+            .chain_state_view(self.id())
+            .await
+            .expect("Failed to read chain state");
```

Comment:
Would it be useful for testing to (in addition or instead of this) also have a method that returns the whole chain state view?

Reply:
I'm not sure if it would be too much of a leaky abstraction to application developers. I think it might make more sense to provide individual methods and/or a GraphQL (like?) query interface. But we can think more about this for a future PR.